terraform/modules/emblem-app/main.tf

```diff
@@ -78,3 +78,102 @@ resource "google_storage_bucket_iam_member" "sessions-iam" {
 }
+
+############################
+# E2E Delivery testing IAM #
+############################
+
+resource "google_project_iam_member" "delivery_gae_viewer_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/appengine.appViewer"
+  member   = "serviceAccount:${data.google_project.ops.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_app_services
+  ]
+}
+
+resource "google_project_iam_member" "delivery_sa_admin_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/iam.serviceAccountAdmin"
+  member   = "serviceAccount:${data.google_project.ops.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_app_services
+  ]
+}
+
+resource "google_project_iam_member" "delivery_compute_viewer_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/compute.viewer"
+  member   = "serviceAccount:${data.google_project.ops.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_app_services
+  ]
+}
+
+resource "google_project_iam_member" "delivery_iam_reviewer_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/iam.securityReviewer"
+  member   = "serviceAccount:${data.google_project.ops.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_app_services
+  ]
+}
+
+resource "google_project_iam_member" "delivery_storage_object_admin_granting_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/iam.securityAdmin"
+  member   = "serviceAccount:${data.google_project.target_project.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_ops_services
+  ]
+
+  condition {
+    title       = "Sessions Object Admin only"
+    description = "Only allow granting/revocation of the Storage Object Admin role to the Sessions bucket"
+    expression  = "api.getAttribute('iam.googleapis.com/modifiedGrantsByRole', []).hasOnly(['roles/storage.objectAdmin']) && resource.type == 'storage.googleapis.com/Bucket' && resource.name.endsWith('sessions')"
+  }
+}
+
+resource "google_project_iam_member" "delivery_run_admin_granting_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/iam.securityAdmin"
+  member   = "serviceAccount:${data.google_project.target_project.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_ops_services
+  ]
+
+  condition {
+    title       = "Cloud Run roles only"
+    description = "Only allow granting/revocation of roles relevant to Cloud Run"
+    expression  = "api.getAttribute('iam.googleapis.com/modifiedGrantsByRole', []).hasOnly(['roles/run.admin', 'roles/iam.serviceAccountUser', 'roles/datastore.user'])"
+  }
+}
+
+resource "google_project_iam_member" "delivery_storage_admin_iam_member" {
```

Comment:
Similar to the above, why not use `google_storage_bucket_iam_member` to bind the permission just to the chosen bucket?

Reply:
I haven't tested that, but we can use this for now.

(Ideally, we can **merge this PR first** and correct any bugs as fast-follows - to avoid having too many moving parts to keep track of! 😬 )